File: tests/fixtures/python/ClassicForMultipleInitUpdate.py
from __future__ import annotations


class ClassicForMultipleInitUpdate:
    """Classic for-loops with multiple initializer and updater expressions."""

    def meet_in_middle(self, len2: int) -> bool:
        i = 1
        j = len2 - 1
        while i <= j:
            if i == j:
                return True
            i += 1
            j -= 1
        return False

    def sum_pair(self, limit: int) -> int:
        total = 0
        left = 0
        right = limit
        while left < right:
            total += left + right
            left += 1
            right -= 1
        return total

    def count_with_assignments(self, limit: int) -> int:
        i = None
        j = None
        seen = 0
        i = 0
        j = limit
        while i < j:
            seen += 1
            i += 1
            j -= 1
        return seen

File: tests/fixtures/java/ClassicForMultipleInitUpdate.java
package com.example;

/** Classic for-loops with multiple initializer and updater expressions. */
public class ClassicForMultipleInitUpdate {

    public boolean meetInMiddle(int len2) {
        for (int i = 1, j = len2 - 1; i <= j; i++, j--) {
            if (i == j) {
                return true;
            }
        }
        return false;
    }

    public int sumPair(int limit) {
        int total = 0;
        for (int left = 0, right = limit; left < right; left++, right--) {
            total += left + right;
        }
        return total;
    }

    public int countWithAssignments(int limit) {
        int i;
        int j;
        int seen = 0;
        for (i = 0, j = limit; i < j; i++, j--) {
            seen += 1;
        }
        return seen;
    }
}
